### DISCUSSION TITLE
How to reveal database schema on datagrip

### DISCUSSION THREAD

**Discussion by MrM4tthew** [cockroachdb/cockroach#71766] [Q&A] [answered]
i successfully connect my db with jetbrains datagrip editor, but when trying to open the schema it show this error
<img width="415" alt="Screen Shot 2021-10-20 at 23 31 43" src="https://user-images.githubusercontent.com/43696102/138133792-0cba7b4e-7628-49ed-8cb0-852a4508efb8.png">
 
i don't really understand what happened, I have used the connection parameters shown on my cluster but it just doesn't show the schema. Funny thing, when I manually query the data like this
<img width="1148" alt="Screen Shot 2021-10-20 at 23 34 19" src="https://user-images.githubusercontent.com/43696102/138134207-3dc2dfbd-05c0-4916-9c06-c3b441a5a054.png">
it shows the table inside the schema. Can somebody help me?

**Comment by jordanlewis** [comment, marked as answer]
You need to make sure to specify your cluster name in your connection string. Could you share the settings you've used to connect? 

Use the settings here to populate in datagrip:

![image](https://user-images.githubusercontent.com/43821/138185283-5070027c-e798-4c07-9f82-4b32f595e560.png)

Like this:

![image](https://user-images.githubusercontent.com/43821/138185319-219ad79b-df59-41b2-8862-cd02d7cfca01.png)
![image](https://user-images.githubusercontent.com/43821/138185338-cb56996b-60cf-478b-9824-0ddb0be36881.png)


Does that help?

**Reply by MrM4tthew** [reply]
i have previously specified the cluster name in my connection string

Just like this:

<img width="565" alt="Screen Shot 2021-10-21 at 09 04 01" src="https://user-images.githubusercontent.com/43696102/138198977-b29e34be-42f6-45f9-88de-f90ac684e87c.png">
<img width="563" alt="Screen Shot 2021-10-21 at 09 04 07" src="https://user-images.githubusercontent.com/43696102/138198630-5ac78223-df46-45b3-b8c1-435471645079.png">

still show the same error, I can CREATE and SELECT the table using the query console but can not manage to visualize it with the datagrip UI. I think the problem is not with me not specifying the cluster name because clearly I have typed it.

**Reply by MrM4tthew** [reply]
@jordanlewis

**Reply by jordanlewis** [reply]
@MrM4tthew, the problem is that JetBrains opens connections to the DB without including the DB in the connection string every time. To resolve this, you should

1. change the database to connect to to just be `defaultdb`
2. set the value of the `options` parameter in the `Advanced` tab to `--cluster=matthew-bennett-355`

This should resolve the problem, please let me know if it works for you.

We will be working on documenting this more clearly.

**Reply by MrM4tthew** [reply]
finally, it works like a charm.
